language-design.md

```diff
@@ -59,7 +83,13 @@ What does the post on grayscale tell us about the process of API design?
 
 **Response**
 
+Lea Verou’s forum post on grayscale is the embodiment of several important design paradigms explored in Bloch’s API design paper. To elaborate, this forum post was created by Lea Verou to ask users to vote on the best name for an API function she was designing. Specifically, the function name was supposed to describe a routine that sets a color to an appropriate shade of gray when provided a lightness and optional alpha parameter.
+ 
+By asking her users to vote on a function name, Verou is clearly in compliance with Bloch’s suggestion that “Names matter,” and “API design is not a solitary activity” [Bloch, 2006, 506]. For instance, if the function names or her users did not matter, why would the developer ever go through the trouble of poling users for something so trivial? It is because, like a DSL, a good API strives to be a simplification of functionality that is intuitive for users. Therefore, a good API should not be what is easiest for the designer to implement and remember, but should conform to the needs of the user base.
 
+Furthermore, assuming Verou will name her function in correspondence to the popular vote, by quantifying the user’s opinions, Verou is also complying with Bloch’s advice that, “You can’t please everyone so aim to displease everyone equally” [Bloch, 2006, 506]. While this advice is a little bit of a joke, the essential take away is that _any_ decision made by a developer will likely benefit one user while inconveniencing another – therefore, since every decision has both good and bad consequences, a developer should strive to benefit the most users while catering to no individual in specific. For instance, none of the proposed function names in this post are illogical. In fact, they’re all _good_ since they make sense in their respective contexts and follow Bloch’s advice of “…the principle of least astonishment,” where every “…method should do the least surprising thing it could, given its name” [Bloch, 2006, 506]. Verou even shows concern for overloading the rgb() and rgba() functions for this purpose similar to Bloch’s advice of  “Overload with care.” [Bloch, 2006, 507]. Therefore, by voting Verou is essentially attempting to displease the fewest number of users, or in other words, “displease everyone equally” [Bloch, 2006, 506].
+
+Ultimately this all boils down to the fact that there are no _objectively right_ name for these functions beyond catering to the users. This lack of altruism, however, is in direct contrast to most other aspects of computer science where a well-defined metric will usually make the decision for the developer. Since API design caters directly to humans and what _feels_ right to them, Bloch’s statement “API design is an art, not a science. Strive for beauty, and trust your gut,” [Bloch, 2006, 507] suddenly makes a lot of sense. This forum post is essentially an art exhibition where Verou is gathering feedback on her craft to better her future work.
```

Comment:
I think it's important to remember that the proposed names are all more or less equal in the eyes of Verou. However, Verou allows users to submit their own function names. By doing so, she can get a lot of user feedback but she runs the risk of discovering a popular name that doesn't fit well with her API. It is possible that a user could suggest a very popular name that is inconvenient to have elsewhere in the API. If this ends up being the case, she would displease the most users by not choosing that name (this might then suggest deeper problems in the overall user-friendliness of the API). I think that this is very unlikely but it is interesting to think about different sides to the approach of polling the user, especially when we do not usually approach computer science in this way.